File: java/src/main/java/com/bilt/pos/nexo/client/BiltNexoClientException.java
/*
 *    ____  _ _ _
 *   | __ )(_) | |_
 *   |  _ \| | | __|
 *   | |_) | | | |_
 *   |____/|_|_|\__|
 *
 *   Bilt POS SDK
 */
package com.bilt.pos.nexo.client;

/**
 * Exception thrown when communication with the Nexo terminal fails.
 *
 * <p>Wraps serialization errors, HTTP transport failures, non-2xx responses,
 * and deserialization errors into a single checked exception.</p>
 */
public class BiltNexoClientException extends Exception {

    public BiltNexoClientException(String message) {
        super(message);
    }

    public BiltNexoClientException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: kotlin/src/main/kotlin/com/bilt/pos/nexo/client/BiltNexoClientException.kt
/*
 *    ____  _ _ _
 *   | __ )(_) | |_
 *   |  _ \| | | __|
 *   | |_) | | | |_
 *   |____/|_|_|\__|
 *
 *   Bilt POS SDK
 */
package com.bilt.pos.nexo.client

/**
 * Thrown when the terminal client encounters an error during
 * serialization, HTTP transport, or deserialization.
 */
class BiltNexoClientException(
    message: String,
    cause: Throwable? = null
) : Exception(message, cause)
